service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -756,6 +759,33 @@ public void deleteBookLog(JWTData userData, int schoolId, int bookId) {
     log.store();
   }
 
+  @Override
+  public SchoolIdListResponse getSchoolsFromUserIdReports(JWTData userData) {
+
+    Set<Integer> ids = new HashSet<>();
+    List<ReportWithLibrary> withLibraryReports =
+        db.selectFrom(SCHOOL_REPORTS_WITH_LIBRARIES)
+            .where(SCHOOL_REPORTS_WITH_LIBRARIES.DELETED_AT.isNull())
+            .and(SCHOOL_REPORTS_WITH_LIBRARIES.USER_ID.eq(userData.getUserId()))
+            .fetchInto(ReportWithLibrary.class);
+
+    List<ReportWithoutLibrary> noLibraryReports =
+        db.selectFrom(SCHOOL_REPORTS_WITHOUT_LIBRARIES)
+            .where(SCHOOL_REPORTS_WITHOUT_LIBRARIES.DELETED_AT.isNull())
+            .and(SCHOOL_REPORTS_WITHOUT_LIBRARIES.SCHOOL_ID.eq(userData.getUserId()))
+            .fetchInto(ReportWithoutLibrary.class);
+
+    for (ReportWithLibrary report: withLibraryReports) {
+      ids.add(report.getSchoolId());
+    }
+
+    for (ReportWithoutLibrary report: noLibraryReports) {
+      ids.add(report.getSchoolId());
+    }
+
+    return new SchoolIdListResponse(ids);
```

Comment:
I'm not sure this produces the desired results. If we're looking at two different tables, IDs can legitimately overlap, meaning there might be a scenario where the user has submitted report with ID 10 for `SCHOOL_REPORTS_WITH_LIBRARIES` but also report with ID 10 for `SCHOOL_REPORTS_WITHOUT_LIBRARIES`.

Let's update this so that we aren't using one set of IDs, but rather two distinct lists (one for each report type)